### PR TITLE
[corlib] Fix RuntimeFeatureTest.NoNewFeaturesAdded on mobile

### PR DESCRIPTION
**Pull request by akoeplinger**
The code uses reflection to access all the fields in the RuntimeFeatures class. When running the linker we strip out these fields since they're `const` and the test fails because it can't find the expected values.

Since the linker always removes all constant fields we can't simply "use" the field in the test and instead we just bail out when encountering this case.

Also used Dictionary instead of tuples for the test since it renders better in the nunit output.

Fixes https://github.com/mono/mono/issues/7145

### CODE REVIEW

**Comment by akoeplinger** [comment]
@monojenkins backport 2018-02

**Comment by akoeplinger** [comment]
@monojenkins backport 2017-12

File: mcs/class/corlib/LinkerDescriptor/mscorlib.xml [hunk 1]
```diff
@@ -676,6 +676,9 @@
 			<method signature="System.Void .ctor(System.Object)" />
 		</type>
 
+		<!-- fields are accessed via reflection by external code -->
+		<type fullname="System.Runtime.CompilerServices.RuntimeFeature" preserve="fields" />
```
Comment (marek-safar) [inline mcs/class/corlib/LinkerDescriptor/mscorlib.xml]: We don't want to always preserve

File: mcs/class/corlib/Test/System.Runtime.CompilerServices/RuntimeFeatureTest.cs [hunk 2]
```diff
@@ -36,22 +37,34 @@ namespace MonoTests.System.Runtime.CompilerServices
...
             var features = from field in t.GetFields()
                 where field.FieldType == typeof (string)
                 let value = field.GetValue (null)
-                select (
+                select new KeyValuePair<string, bool> (
                     field.Name,
                     RuntimeFeature.IsSupported ((string)value)
                 );
-            Assert.AreEqual (ExpectedFeatures, features.ToArray ());
+            CollectionAssert.AreEquivalent (ExpectedFeatures, features.ToDictionary (k => k.Key, v => v.Value));
```
Comment (marek-safar) [inline mcs/class/corlib/Test/System.Runtime.CompilerServices/RuntimeFeatureTest.cs]: Add Assert.Inconclusive when features are empty

**Pull request merged**